Core/src/Vertexing/GaussianTrackDensity.cpp

```diff
@@ -58,15 +59,25 @@ std::pair<double, double> Acts::GaussianTrackDensity::globalMaximumWithWidth(
                       maxDensity, maxSecondDerivative);
   }
 
-  return (maxSecondDerivative == 0.)
-             ? std::make_pair(0., 0.)
-             : std::make_pair(maxPosition,
-                              std::sqrt(-(maxDensity / maxSecondDerivative)));
+  if (maxSecondDerivative == 0.) {
+    return std::nullopt;
+  }
+
+  return std::make_pair(maxPosition,
+                        std::sqrt(-(maxDensity / maxSecondDerivative)));
```

Comment:
NIT, this with deduction guides, I think this can be 
```suggestion
  return std::pair{maxPosition,
                        std::sqrt(-(maxDensity / maxSecondDerivative))};
```